lib/trento_web/controllers/cluster_controller.ex

```diff
@@ -55,4 +55,34 @@ defmodule TrentoWeb.ClusterController do
         |> json(%{error: reason})
     end
   end
+
+  @spec get_connection_settings(Plug.Conn.t(), map) :: Plug.Conn.t()
+  def get_connection_settings(conn, %{"cluster_id" => cluster_id}) do
+    settings = Clusters.get_hosts_connection_settings(cluster_id)
+
+    conn
+    |> put_status(:ok)
+    |> json(settings)
+  end
+
+  @spec save_connection_settings(Plug.Conn.t(), map) :: Plug.Conn.t()
+  def save_connection_settings(
+        conn,
+        %{
+          "settings" => [_ | _] = settings,
+          "cluster_id" => cluster_id
+        }
+      ) do
+    settings
+    |> Enum.map(&map_to_struct/1)
+    |> Clusters.save_hosts_connection_settings()
+
+    get_connection_settings(conn, %{"cluster_id" => cluster_id})
+  end
+
+  defp map_to_struct(map) do
+    for {key, val} <- map, into: %{} do
+      {String.to_atom(key), val}
```

Comment:
As far as I know this `String.to_atom` is a non go. It create some vulnerability issues:
http://www.moxleystratton.com/when-to-use-elixirs-string-to-atom/

In any case, I don't really understand why we need this `map_to_struct` method

Reply:
I needed that because what comes in from an http request is a map
```
[
  %{"host_id" => "13e8c25c-3180-5a9a-95c8-51ec38e50cfc", "user" => "pippo"},
  %{"host_id" => "0a055c90-4cb6-54ce-ac9c-ae3fedaf40d4", "user" => nil}
]
```

and I want to the inner layers to have structs
```
[
  %{host_id: "13e8c25c-3180-5a9a-95c8-51ec38e50cfc", user: "pippo"},
  %{host_id: "0a055c90-4cb6-54ce-ac9c-ae3fedaf40d4", user: nil}
]
```

Gonna do that without using `String.to_atom`

Reply:
changed to 
```
defp map_to_struct(%{"host_id" => host_id, "user" => user}) do
  %{host_id: host_id, user: user}
end
```

Reply:
Most probably @fabriziosestito has more insights no this in any case hehe